src/sd/mmc.rs

```diff
@@ -0,0 +1,448 @@
+use core::marker::PhantomData;
+use core::ops::Deref;
+use core::sync::atomic::{AtomicU8, Ordering};
+
+use crate::gpio::{self, InputPin, OutputPin};
+use crate::peripheral::Peripheral;
+use crate::sys::*;
+
+/// Indicates that card detect line is not used
+const SDMMC_SLOT_NO_CD: i32 = -1;
+/// Indicates that write protect line is not used
+const SDMMC_SLOT_NO_WP: i32 = -1;
+/// Bit indicating that internal pullups should be enabled
+const SDMMC_INTERNAL_PULLUPS_ENABLE_FLAG: u32 = 1;
+
+static USED_SLOTS: AtomicU8 = AtomicU8::new(0);
+static USED_SLOTS_CS: crate::task::CriticalSection = crate::task::CriticalSection::new();
+
+pub type SdMmcHostConfiguration = config::Configuration;
+
+pub mod config {
+    /// Configuration for the SD-MMC Host driver
+    #[non_exhaustive]
```

Comment:
Why do you want this to be non_exhaustive? I understand in the other places where we should add them and we know upstream changes stuff that we need it. Though in this case "sd" is behind a experimental flag so i think early upgrades to Configurations are fine here if you need them and if it ever reached a point of stability, removing a non_exhaustive is also a breaking change right?

Overall i am fine with it being non_exhaustive i just want to understand your reason here.

Reply:
Think about it this way: if I don't put non_exhaustive NOW, what would that mean? That we should not forget to put it later. On the other hand, putting it now does absolutely no harm.

Reply:
Non_exhaustive does NOT mean we can't do "early upgrades". We sure can.